### PR TITLE
Cleanup logging and str formatting

### PR DESCRIPTION
**Pull request by jenshnielsen**
Only use logformatting or f strings to format logs. Ideally we should only use log formatting and not f strings for performance but this is at least a step in the right direction.

Use ruff to enforce this.

Also fix and enforce [implicit-str-concat-isc](https://beta.ruff.rs/docs/rules/#flake8-implicit-str-concat-isc)

### CODE REVIEW

File: qcodes/instrument_drivers/signal_hound/SignalHound_USB_SA124B.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def set_sweep(self, sweep_len: int, start_freq: float, stepsize: float) -> None:
         """
         if not isinstance(self.instrument, SignalHoundUSBSA124B):
             raise RuntimeError(
-                "'FrequencySweep' is only implemented" "for 'SignalHound_USB_SA124B'"
+                "'FrequencySweep' is only implementedfor 'SignalHound_USB_SA124B'"
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/signal_hound/SignalHound_USB_SA124B.py]: Missing whitespace

File: qcodes/instrument_drivers/tektronix/AWG5014.py [hunk 2]
```diff
@@ -440,8 +438,7 @@ def get_state(self) -> Literal['Idle', 'Waiting for trigger', 'Running']:
         elif state.startswith('2'):
             return 'Running'
         else:
-            raise ValueError(__name__ + (' : AWG in undefined ' +
-                                         'state "{}"').format(state))
+            raise ValueError(f'__name__ : AWG in undefined state "{state}"')
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/tektronix/AWG5014.py]: Missing brackets

File: qcodes/parameters/multi_parameter.py [hunk 3]
```diff
@@ -176,7 +176,7 @@ def __init__(
 
         if not is_sequence_of(shapes, int, depth=2) or len(shapes) != len(names):
             raise ValueError(
-                "shapes must be a tuple of tuples " "of ints, not " + repr(shapes)
+                "shapes must be a tuple of tuples of ints, not " + repr(shapes)
```
Comment (jenshnielsen) [inline qcodes/parameters/multi_parameter.py]: Put inline

File: qcodes/utils/helpers.py [hunk 4]
```diff
@@ -37,8 +37,10 @@
 # on longer in used but left for backwards compatibility until
 # module is removed.
 def warn_units(class_name: str, instance: object) -> None:
-    logging.warning('`units` is deprecated for the `' + class_name +
-                    '` class, use `unit` instead. ' + repr(instance))
+    logging.warning(
+        f"`units` is deprecated for the ` {class_name} "
+        f"` class, use `unit` instead. {repr(instance)}"
+    )
```
Comment (jenshnielsen) [inline qcodes/utils/helpers.py:43]: Check quotes

**Review by astafan8 — APPROVED** [review]
nice cleanup! ruff rocks :)

**Pull request merged**